src/Development/IDE/Core/Shake.hs

```diff
@@ -104,6 +105,10 @@ data ShakeExtras = ShakeExtras
     -- accumlation of all previous mappings.
     ,inProgress :: Var (HMap.HashMap NormalizedFilePath Int)
     -- ^ How many rules are running for each file
+    ,getLspId :: IO LspId
+    ,reportProgress :: Bool
```

Comment:
Can we get comments on these fields - I find these some of the most useful comments in the code base.

Reply:
done